internal/overrides/handler.go

```diff
@@ -0,0 +1,115 @@
+package overrides
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"gopkg.in/yaml.v3"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type Handler struct {
+	client       client.Reader
+	config       HandlerConfig
+	atomicLevel  zap.AtomicLevel
+	defaultLevel zapcore.Level
+}
+
+type HandlerConfig struct {
+	ConfigMapName types.NamespacedName
+	ConfigMapKey  string
+}
+
+func New(client client.Reader, atomicLevel zap.AtomicLevel, config HandlerConfig) *Handler {
+	return &Handler{
+		atomicLevel:  atomicLevel,
+		defaultLevel: atomicLevel.Level(),
+		client:       client,
+		config:       config,
+	}
+}
+
+func (h *Handler) LoadOverrides(ctx context.Context) (*Config, error) {
+	overrideConfig, err := h.loadOverrides(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load overrides config: %w", err)
+	}
+
+	if err := h.syncLogLevel(ctx, overrideConfig.Global); err != nil {
+		return nil, fmt.Errorf("failed to sync log level: %w", err)
+	}
+
+	return overrideConfig, nil
+}
+
+func (h *Handler) loadOverrides(ctx context.Context) (*Config, error) {
```

Comment:
can we may be change the name to eg. `loadOverridesConfig` just to avoid confusion with `LoadOverrides` ?